internal/store/device.go

```diff
@@ -155,6 +155,9 @@ func (s *DeviceStore) CreateOrUpdate(ctx context.Context, orgId uuid.UUID, resou
 	device := model.NewDeviceFromApiResource(resource)
 	device.OrgID = orgId
 
+	// Use the dedicated API to update annotations
+	device.Annotations = nil
+
```

Comment:
Why do we clear all annotations here?

Reply:
The user shouldn't set them, so we clear them.  The service uses a dedicated API to set annotations.

Reply:
Right now we use annotations only for the service to annotate a resource for its own purposes.  If a user changed one of those annotations, things would break.  I'm happy to discuss changing that, but it's beyond the scope of this PR.

Reply:
Let's leave it for now. My worry is that this may need to change i.e. if we exposed devices to external controllers, like a fleet manager but made in a different way..